compiler/testData/diagnostics/tests/regressions/kt557.fir.kt

```diff
@@ -8,5 +8,5 @@ fun Array<String>.length() : Int {
 }
 
 fun test(array : Array<String?>?) {
-    array?.sure<Array<String?>>().<!INAPPLICABLE_CANDIDATE!>length<!>()
+    array?.sure<Array<String?>>().<!UNSAFE_CALL!>length<!>()
```

Comment:
To be honest, no idea yet.

https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/regressions/kt557.kt#L11

At DOT operator, the receiver has `TYPE_MISMATCH`, and the call is reported as `UNRESOLVED_REFERENCE_WRONG_RECEIVER` (well, because type mismatches, hm). We don't have a type checker that reports `TYPE_MISMATCH`, don't we?